File: go32/adt2vesa/pas/VESA_StepFade.pas
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2024 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

procedure VESA_StepFade;

var
  idx: Word;

begin
  For idx := 0 to 767 do
    begin
      StepRealPal[idx] := StepRealPal[idx]+StepDelta[idx];
      StepWorkPalette[idx] := ROUND(StepRealPal[idx]);
    end;
  VESA_SetPalette(StepWorkPalette);
end;

File: go32/adt2vesa/VESA_StepFade.c
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2024 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

void VESA_StepFade (void) {
  uint_least16_t i;

  for (i = 0; i < 256*3; i++) {
      StepRealPal[i] += StepDelta[i];
      StepWorkPalette[i] = (uint8_t) StepRealPal[i];
  }

  VESA_SetPalette (StepWorkPalette);
}
